flutter-studio/src/io/flutter/actions/ConnectAndroidDebuggerAction.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2018 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.actions;
+
+import com.google.common.base.Joiner;
+import com.intellij.execution.Executor;
+import com.intellij.execution.ProgramRunnerUtil;
+import com.intellij.execution.RunnerAndConfigurationSettings;
+import com.intellij.execution.configurations.RunConfiguration;
+import com.intellij.execution.runners.ExecutionEnvironment;
+import com.intellij.execution.runners.ExecutionEnvironmentBuilder;
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.application.ModalityState;
+import com.intellij.openapi.ui.DialogWrapper;
+import com.intellij.openapi.ui.Messages;
+import com.intellij.openapi.wm.ToolWindowId;
+import io.flutter.FlutterInitializer;
+import io.flutter.FlutterUtils;
+import io.flutter.run.FlutterLaunchMode;
+import io.flutter.run.SdkFields;
+import io.flutter.run.SdkRunConfig;
+import io.flutter.run.attach.SdkAttachConfig;
+import org.jetbrains.android.actions.AndroidConnectDebuggerAction;
+import org.jetbrains.annotations.Nullable;
+
+import javax.swing.*;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ConnectAndroidDebuggerAction extends AndroidConnectDebuggerAction {
+  @SuppressWarnings("FieldCanBeLocal")
+  private static String RUN_DEBUG_LINK = "https://flutter.io/using-ide/#running-and-debugging";
+
+  @Override
+  public void actionPerformed(AnActionEvent e) {
```

Comment:
If this method is copied from another (which I'm assuming it is given the `--flavor` and other code below), can you add comments in this method and the source method that the two should be kept in sync? Otherwise we risk fixes and changes in one not being applied to the other.

Reply:
Will do.